lambda_functions/alerter/alerter.py

```diff
@@ -0,0 +1,69 @@
+import os # isort:skip
+import json # isort:skip
+import urllib.request # isort:skip nosec
+
+def lambda_handler(event, context):
+    """
+    lambda_handler The purpose of this lambda function is to alert the sml-portal-alert
+    slack channel if an alarm is failing (failing alarm triggers lambda)
+
+    :param event: Event is the message contents parsed to the lambda
+    :type event: string
+    :param context: General Lambda term (not used in this case but needed for general setup)
+    :type context: N/A
+    :return: Return a success or failure outcome on whether the slack message was sent
+    :rtype: json
+    """    
+    
+    # To make the lambda reusable we have the option of parsing event data
+    # If no event data is provided we default to environment variables
+    if 'lambda_name' in event:
+        lambda_name = event['lambda_name']
+    else:
+        lambda_name = os.environ.get('lambda_name')
+    
+    if 'alarm_name' in event:
+        alarm_name = event['alarm_name']
+    else:
+        alarm_name = os.environ.get('alarm_name')
+    
+    if 'url' in event:
+        url = event['url']
+    else:
+        url = f"https://{os.environ.get('url')}"
+
+    # Slack channel link for alerting purposes
+    if 'slack_webhook_url' in event:
+        slack_webhook_url = event['slack_webhook_url']
+    else:
+        slack_webhook_url = f"{os.environ.get('slack_webhook_url')}"
+
+    # Message sent to channel
+    alert_message = {
+        "Summary": f"The website at {url} is unreachable. Healthcheck failure.",
+        "Alarm": "Route53 Health Check Failure",
+        "Description": f"Website is unreachable or not returning the expected response. Check Amazon Cloudwatch Alarms \'{alarm_name}\' and Healthcheck Lambda \'{lambda_name}\'.",
+    }
+
+    # Prepare the request
+    request = urllib.request.Request( # nosec
```

Comment:
The requests libary does not work with the modern versions of python in aws lambda i tried using lambda layers to get around this but could only find a way around it using urlib. We could downgrade the version of python but then the stack would not deploy and concourse would fail.

Reply:
reverted back to requests